UX_PRINCIPLES.md

```diff
@@ -0,0 +1,79 @@
+- Write output with a clear messages that a) sets the stage with “what we’re about to do”, b) progress “what is happening now”, and c) “did it fail or succeed?” and if it failed explain why, then provide a recommendation/next step. If it succeeds, celebrate it with memorable/playful message.
+### Expose Your Personality
+- It’s a machine, but it doesn’t have to feel like one. Remind your users that there are people behind this product and that they care about your experience.
+- Don’t overuse this principle by being overly silly or ever-present (that will be annoying), but a couple of inside jokes or playfully-worded sentences (especially during moments of success) sprinkled throughout keeps things interesting and personable (e.g. when a plan builds successfully out last line of output is “I love it when a plan.sh comes together.”)
+### Provide Feedback
+- Provide a sense of how long something will take if it’s going to longer than a few seconds with progress bars and/or honest message like “this may take several minutes” and show that it is progressing (not stuck)
+### Avoid Dead-ends
+- Error message format / help links and instructions
+
+## Instrumentation
+### Measure Effectiveness
+- It’s the command line so you can’t get any analytics, right? Wrong. As with a web or mobile application, you can still implement tracking with common tools like Google Analytics. That said, you have some unique challenges to address:
+ - Treat the terminal experience as a more private and sacred interface for your users. Unlike the app world where there is obvious reliance upon the internet and thus an implied notion of less privacy, the command line experience feels more confined to the walls of your machine.
+ - We’ve seen examples of community backlash when analytics were opted in by default (see homebrew) and it was not pretty. Given this more private setting, be courteous and transparent to retain trust and get the data you need. Some tips on accomplishing include:
+  - Set analytics as opt out by default unless you are absolutely certain (and have communicated thoroughly elsewhere) that you’re tracking things.
+  - Be explicit on what/when/why/how your sending data. Put it in your permission request test, code comments, and docs. Be transparent; have nothing to hide.
+  - Find the right time to request permission. For example, provide some helpers for getting up and running (see ‘Be a Good Host’ above) and in return, after having helped them, ask for participation in your analytics program.
+  - Do not send any sort of personally identifiable, private or sensitive data.
+   - Outline every data point that you intend to capture and send.
+   - Once it has been sent, leave no trace of that history.
+  - Avoid creating delays in the core experience. It goes without saying, but your users are here to accomplish a task and that is paramount to your data needs. Don’t optically degrade the UX to serve yourself and your analytical desires.
```

Comment:
It won't block them, but you'd see the attempt to send mixed in with the output. I'll add this additional clarification.